src/h2/stream.py

```diff
@@ -1319,7 +1322,7 @@ def _initialize_content_length(self, headers):
                     self._expected_content_length = int(v, 10)
                 except ValueError:
                     raise ProtocolError(
-                        "Invalid content-length header: %s" % v
+                        f"Invalid content-length header: {repr(v)}"
```

Comment:
This (and the family of these) are not unrelated. Since all header name and values are now bytes, they need explicit casting to `str`. Using `repr()` gives us a better representation so I opted for that.

The only thing we can argue is the usage of f-strings. For that, I'd say since we are touching this line, it is fair game to switch the old `%` syntax to `f-strings` for readability. Would revert that part if you insist tho.

Reply:
Thanks for the explanation - makes sense now to have this change!